src/Microsoft.Identity.Client/Utils/Maybe.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Identity.Client.Utils
+{
+    internal sealed class Maybe<T>
+    {
+        internal bool HasItem { get; }
+        internal T Item { get; }
+
+        public Maybe()
+        {
+            HasItem = false;
+        }
+
+        public static Maybe<T> Empty()
+        {
+            return new Maybe<T>();
```

Comment:
>new Maybe<T>(); [](start = 19, length = 15)

not happy to say this, however should this be returning the same instance (static)?

Reply:
I've overriden Equals and GetHashCode so 2 empty Maybes will be equal for all intents and purposes. 

However note that I've removed this class.